org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java

```diff
@@ -139,64 +149,30 @@ public static String readFileIntoString(String filename) {
 	
 	/**
 	 * @since 2.3
+	 * @deprecated Use com.google.common.io.ByteStreams.toByteArray(InputStream) instead
 	 */
+	@Deprecated
 	public static byte[] readStreamIntoByteArray(InputStream inputStream) {
 		try {
-			ByteArrayOutputStream buffer = new ByteArrayOutputStream();
-	
-			int nRead;
-			byte[] data = new byte[16384];
-	
-				while ((nRead = inputStream.read(data, 0, data.length)) != -1) {
-				  buffer.write(data, 0, nRead);
-				}
-	
-			buffer.flush();
-	
-			return buffer.toByteArray();
+			return ByteStreams.toByteArray(inputStream);
```

Comment:
Please add buffering to the inputStream if it is not yet buffered

Reply:
toByteArray calls copy(in,out), which does buffering
com.google.common.io.ByteStreams.copy(InputStream, OutputStream)

Am I missing something?

Reply:
I suspect that a buffered input stream with a buffer size of 16k is the better choice than a custom buffering in ByteStreams.copy with a buffer size of 4k. But that is only an assumption that would need to be verified.

Reply:
Actually there are no references to this method. Also an indicator to deprecate it?

Reply:
Yes